frontend/gateway/gateway.go

```diff
@@ -656,16 +696,20 @@ func (lbf *llbBridgeForwarder) Return(ctx context.Context, in *pb.ReturnRequest)
 	}
 }
 
-func (lbf *llbBridgeForwarder) convertRef(id string) (solver.CachedResult, error) {
-	if id == "" {
+func (lbf *llbBridgeForwarder) convertRef(ids []string) (solver.CachedResult, error) {
+	if len(ids) == 0 {
 		return nil, nil
 	}
+
```

Comment:
I'm assuming this is where the error checking for multi-result array to be? This is where the forwarder is handling the `Return` RPC from the frontend.